src/outcomes.py

```diff
@@ -5,26 +5,38 @@
 UnionArray = Union[np.ndarray, pd.DataFrame]
 
 
-# Class to store the results of the voting,
-# like a dictionary but with a winner field
 class Result(dict):
+    """''
+    # Class to store the results of the voting,
+    # like a dictionary but with a winner field
+    """
+
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
 
-    # Method to get the winner of the voting, resolving ties by
-    # returning the first one in alphabetical order
     @property
     def winner(self) -> str:
-        return max(self, key=self.get)
+        """
+        Method to get the winner of the voting, resolving ties by
+        returning the first one in alphabetical order
+        """
+        max_value = max(self.values())
+        keys_max_value = [k for k in self.keys() if self.get(k) == max_value]
+        return min(keys_max_value)
 
-    # Overriding the __repr__ method to include the winner
     def __repr__(self):
+        """
+        Overriding the __repr__ method to include the winner
+        """
         return f"Winner: {self.winner}\n{super().__repr__()}"
 
 
-# Decorator to wrap the outcome functions, to accept both numpy arrays and
-# pandas dataframes and return a Result object
 def outcome_wrapper(func):
+    """
+    Decorator to wrap the outcome functions, to accept both numpy arrays and
+    pandas dataframes and return a Result object
+    """
```

Comment:
We can simplify the wrapper code. In the outcome functions instead of writing, `result = dict(zip(alternatives, votes))` we can write `result = Result(zip(alternatives, votes))`. As `Result` inherits from `dict`. Then we can remove the code from `L48` to `L51`.

Reply:
I got what you are saying and i am now returning `Result` types of object, but the code also adds all the remaining alternatives with 0 votes, that sometimes might get discarded. The code should now be more clear